src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,291 @@
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        success = False
+
+        if not self.connection_is_activated(connection_id):
+            connection = self.connection_id_to_connection_dict[connection_id]
+            network_device = self._get_device_for_connection(connection)
+
+            networkmanager_output = NetworkManager.NetworkManager.ActivateConnection(
+                connection, network_device, '/')
+
+            try:
+                self._run_proxy_call(networkmanager_output)
+                success = self._wait_for_connection(connection)
+
+            except Exception as exception:
+                self.logger.error('D-Bus call failed: %s', exception)
+
+        else:
+            self.logger.debug('Connection %s is already activated.', connection_id)
+            success = True
+
+        return success
+
+    def get_connection_ip(self, connection_id):
+        """Attempts to retrieve the IP address associated with the given connection's gateway.
+        If the IP address is unable to be retrieved, None is returned.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        # I decided not to throw an exception here because the intended caller would end up
+        #   simply using it for flow control.
+        ip_address = None
+
+        connection = self.connection_id_to_connection_dict[connection_id]
+
+        if self.connection_is_activated(connection_id):
+            device = self._get_device_for_connection(connection)
+            gateway_octets = device.Ip4Config.Gateway.split('.')
+
+            for address_data in device.Ip4Config.AddressData:
+                address_octets = address_data['address'].split('.')
+                if address_octets[0] == gateway_octets[0] \
+                        and address_octets[1] == gateway_octets[1] \
+                        and address_octets[2] == gateway_octets[2]:
+
+                    ip_address = address_data['address']
+
+            if ip_address is None:
+                self.logger.warning(
+                    'No IP addresses for connection %s associated with a gateway.',
+                    connection_id)
+
+        else:
+            self.logger.warning(
+                'Attempted to get IP address for connection %s, which is not connected.',
+                connection_id)
+
+        return ip_address
+
+    def connection_is_activated(self, connection_id):
+        """Check whether the network with the given connection id is ready.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+
+        connection_is_activated = False
+
+        connection_state = self._get_connection_state(connection_id)
+
+        if connection_state is NM_CONNECTION_ACTIVATED:
+            connection_is_activated = True
+
+        return connection_is_activated
+
+    def _build_connection_id_table(self):
```

Comment:
This method isn't building a table. I'd rename this method to better reflect what it does.